### PR TITLE
Elixir styleguide draft

### PR DESCRIPTION
**Pull request by maabernethy**

### CODE REVIEW

File: elixir.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+```
+
+* Use parentheses in function calls when passing parameters, especially inside a pipeline.
+
+```
+# bad
+convert_to_celsius 32
+
+# REALLY bad
+# parses as convert_to_celsius(32 |> format)
```
Comment (danmcclain) [inline elixir.md]: Like the *really* bad

File: elixir.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+1.  [Naming](#naming)
+
+## Whitespace
+
+* Use spaces around operators, after commas, colons and semicolons.
+
+* Do not put spaces around matched pairs like brackets, parentheses, etc.
+
+```
+sum = 1 + 2
```
Comment (bcardarella) [inline elixir.md]: should we have examples of the "Do not" code?

File: elixir.md [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+end
+
+def sum([head | tail]) do
+  head + sum(tail)
+end
+```
+
+## Commenting
+
+* Single line comments start with a number symbol.
```
Comment (bcardarella) [inline elixir.md]: I'm not sure if this is so much a styleguide thing as something that is just part of the syntax of the language

File: elixir.md [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+def sum([]), do: 0
+def sum([head | tail]) do
+  head + sum(tail)
+end
+```
+
+* ...however, if you have more than one multi-line function `def`, do not put them together.
+
+```
+def sum(nil) do
```
Comment (danmcclain) [inline elixir.md]: These are bad examples below, we probably shouldn't use them as a good example

File: elixir.md (hunk 2 above)
Comment (danmcclain) [inline elixir.md]: :+1: For consistency, we should always have good and bad examples (the examples below have both)

File: elixir.md [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+
+```
+def sum(nil), do: {:error, "No Value"}
+def sum([]), do: 0
+def sum([head | tail]) do
+  head + sum(tail)
+end
+```
+
+* ...however, if you have more than one multi-line function `def`, do not put them together.
```
Comment (chrismccord) [inline elixir.md]: I group mine multi-line defs of the same name together pretty often, so I would ditch this one

File: elixir.md [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+String.strip(some_string)
+|> String.downcase
+|> String.codepoints
+
+# good
+some_string
+|> String.strip
+|> String.downcase
+|> String.codepoints
+```
```
Comment (chrismccord) [inline elixir.md]: Let's add this to the Pipeline section:

When assigning the result of a pipeline to a variable, the `variable =` should stand on its own, with an indented pipeline below, for example:

```elixir
# bad
result = some_string
|> String.strip
|> String.downcase
|> String.codepoints

# good
result = 
  some_string
  |> String.strip
  |> String.downcase
  |> String.codepoints
```

**Comment by chrismccord** [comment]
Thanks @maabernethy. I tweaked a couple sections and extended a couple areas. Anyone feel free to chime in and voice opinions.

File: elixir.md (hunk 6 above)
Comment (bcardarella) [inline elixir.md]: :+1: I've been doing the former but I like the later better

Comment (chrismccord) [inline elixir.md]: I was the same, but José convinced me on the latter, same with the null arity local functions

**Pull request merged**